Add the router priority documentation (#5481)
containous/traefik: docs/content/routing/routers/index.md

    For instance, `PathPrefix: /products` would match `/products` but also `/products/shoes` and `/products/shirts`.
    Since the path is forwarded as-is, your service is expected to listen on `/products`.

### Priority

> To avoid path overlap, routes are sorted, by default, in descending order using rules length: the largest length has the highest priority.

Reviewer: ```suggestion
To avoid path overlap, routes are sorted, by default, in descending order using rules length. The priority is directly equal to the length of the rule, and so the longest length has the highest priority.
```

> By default, to avoid path overlap, routes will be sorted in descending order using rules length (the largest length has the highest priority).

Reviewer 2: ```suggestion
To avoid path overlap, routes are sorted, by default, in descending order using rules length: the largest length has the highest priority.
```

Reviewer 2: - Present tense
- Make the explanation part of the sentence

> A priority value of 0 will be ignored, so the default value will be calculated (rules length).

Reviewer 2: ```suggestion
A value of `0` for the priority is ignored: `priority = 0` means that the default rules length sorting is used.
```

> ??? info "Calculate default priorities"

Reviewer 3: ```suggestion
??? info "Calculated default priorities"
```

Reviewer 3: Though I'm not sure

Reviewer 4: `Calcute` without `d`

    ```toml tab="File (TOML)"
    ## Dynamic configuration
    [http.routers]
      [http.routers.Router-1]
        rule = "HostRegexp(`.*\.traefik\.com`)"
        # ...
      [http.routers.Router-2]
        rule = "Host(`foobar.traefik.com`)"
        # ...
    ```
    
    ```yaml tab="File (YAML)"
    ## Dynamic configuration
    http:
      routers:
        Router-1:
          rule: "HostRegexp(`.*\.traefik\.com`)"
          # ...
        Router-2:
          rule: "Host(`foobar.traefik.com`)"
          # ...
    ```
    
    In this case, all requests with host `foobar.traefik.com` will be routed through `Router-1` instead of `Router-2`.
    

>     | Name     | Rule                                 | length |

Reviewer: ```suggestion
    | Name     | Rule                                 | priority ( == length) |
```
